docs/client-libraries.md

```diff
@@ -13,220 +13,252 @@ nav = [
+
+# Let's create Index and Frame objects, which will contain the settings
+# for the corresponding indexes and frames.
+try:
+    schema = client.schema()
+except PilosaError as e:
+    # Most calls will raise an exception on errors.
+    # You should handle them appropriately.
+    # We will just terminate the program in this case.
+    raise SystemExit(e)
+
+# We need to refer to indexes and frames before we can use them in a query.
+repository = schema.index("repository")
+stargazer = repository.frame("stargazer")
+language = repository.frame("language")
+
 # Which repositories did user 8 star:
-response = client.query(stargazer.bitmap(8))
-if response.result:
-    print("User 8 starred: ", result.bitmap.bits)
+repository_ids = client.query(stargazer.bitmap(14)).result.bitmap.bits
+print("User 8 starred: ", repository_ids)
 
 # What are the top 5 languages in the sample data:
-response = client.query(language.topn(5))
-if response.result:
-    print("Top 5 languages: ", result.bitmap.bits)
-
-# Which repositories were starred by user 8 and 18:
-response = client.query(
-        repository.intersect(
-            stargazer.bitmap(8),
-            stargazer.bitmap(18)))
-if response.result:
-    print("Repositories starred by both user 8 and 18: ", result.bitmap.bits)
-
-# Which repositories were starred by user 8 and 18 and also were written in language 1
-response = client.query(
-        repository.intersect(
-            stargazer.bitmap(8),
-            stargazer.bitmap(18),
-            language.bitmap(1)))
-if response.result:
-    print("Repositories starred by both user 8 and 18 and are in language 1: ", result.bitmap.bits)
+top_languages = client.query(language.topn(5)).result.count_items
+print("Top 5 languages: ", [item.id for item in top_languages])
 
-# Set user 99999 as a stargazer for repository 77777
-try:
-    client.query(stargazer.setbit(99999, 77777))
-except PilosaError as ex:
-    print("Error setting bit: ", ex)
+# Which repositories were starred by both user 14 and 19:
+query = repository.intersect(
+    stargazer.bitmap(14),
+    stargazer.bitmap(19)
+)
+mutually_starred = client.query(query).result.bitmap.bits
+print("Both user 14 and 19 starred:", mutually_starred)
 
-```
+# Which repositories were starred by user 14 or 19:
+query = repository.union(
+    stargazer.bitmap(14),
+    stargazer.bitmap(19)
+)
+either_starred = client.query(query).result.bitmap.bits
+print("User 14 or 19 starred:", either_starred)
+
+# Which repositories were starred by user 14 or 19 and were written in language 1:
+query = repository.intersect(
+    repository.union(
+        stargazer.bitmap(14),
+        stargazer.bitmap(19)
+    ),
+    language.bitmap(1)
+)
+mutually_starred = client.query(query).result.bitmap.bits
+print("User 14 or 19 starred, written in language 1:", mutually_starred)
+
+# Set user 99999 as a stargazer for repository 77777
+client.query(stargazer.setbit(99999, 77777))`
```

Comment:
This Python block is not properly closed. Needs 3 tick marks.

Reply:
Thanks @codysoyland, updated.

Reply:
No problem. Note there is still a single tick mark at the end of line 169.

Reply:
Updated, hopefully this time I got it right.